src/pool/rpc.h

```diff
@@ -37,8 +37,9 @@
 		0, &CQF_pool_create,					\
 		ds_pool_create_handler, NULL),				\
 	X(POOL_CONNECT,							\
-		0, &CQF_pool_connect,					\
-		ds_pool_connect_handler, NULL),				\
+	  0, ver == 4 ? &CQF_pool_connect_v4 : &CQF_pool_connect_v5,	\
+	  ver == 4 ? ds_pool_connect_handler_v4 :			\
+	  ds_pool_connect_handler_v5, NULL),				\
```

Comment:
I do find myself wondering how we'll handle this if there are more versions added in the future. These macros could get pretty ugly.

Reply:
Yes i agreed, but a second think is i think we are going to maintain two versions, that might be fine.